akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -21,9 +21,25 @@ import scala.concurrent.{ ExecutionContext, Future }
 /**
  * Represents one direction of an Http2 substream.
  */
-private[http2] final case class Http2SubStream(
+private[http2] sealed trait Http2SubStream {
+  type T
```

Comment:
I think the type can be removed if you make it a `val data: Source[_, Any]` below. I think the generic part is only used for cancellation.